src/core/lib/OrderCombiner.sol

```diff
@@ -1178,4 +1185,27 @@ contract OrderCombiner is OrderFulfiller, FulfillmentApplier {
                 )
         }
     }
+
+    /**
+     * @dev Internal pure function to determine whether a status update failure
+     *      should cause a revert or allow a skipped order. The call must revert
+     *      if an authorizeOrder call has been successfully performed and the
+     *      status update cannot be performed, regardless of whether the order
+     *      could be otherwise marked as skipped.
+     *
+     * @param orderType       The order type.
+     * @param revertOnInvalid A boolean indicating whether the call should
+     *                        revert for non-restricted order types.
+     *
+     * @return revertOnFailedUpdate A boolean indicating whether the order should
+     *                              revert on a failed status update.
+     */
+    function _revertOnFailedUpdate(
+        OrderType orderType,
+        bool revertOnInvalid
+    ) internal pure returns (bool revertOnFailedUpdate) {
+        assembly {
+            revertOnFailedUpdate := or(revertOnInvalid, gt(orderType, 1))
```

Comment:
Just explaining in my own words to check my understanding: we want to always provide `revertOnInvalid` to `_updateStatus` because at this point, for restricted (and contract, technically) orders, we already called the pre-hook, which could have modified state. If we fail updating the status (eg order was already filled or cannot apply the supplied fraction) for an order in the `fulfillAvailable` case, we want to revert rather than simply mark as "unavailable" and continue, because the `authorizeOrder` call is stateful, and we won't call the complementary `validateOrder` call for "unavailable" orders (and the supplied spent/received items would not technically be correct in either case).